rp235x-hal/src/arch.rs

```diff
@@ -73,43 +101,476 @@ mod inner {
+                    }
+                    unsafe {
+                        UART1_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::ADC_IRQ_FIFO => {
+                    extern "Rust" {
+                        fn ADC_IRQ_FIFO();
+                    }
+                    unsafe {
+                        ADC_IRQ_FIFO();
+                    }
+                }
+                rp235x_pac::Interrupt::I2C0_IRQ => {
+                    extern "Rust" {
+                        fn I2C0_IRQ();
+                    }
+                    unsafe {
+                        I2C0_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::I2C1_IRQ => {
+                    extern "Rust" {
+                        fn I2C1_IRQ();
+                    }
+                    unsafe {
+                        I2C1_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::OTP_IRQ => {
+                    extern "Rust" {
+                        fn OTP_IRQ();
+                    }
+                    unsafe {
+                        OTP_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::TRNG_IRQ => {
+                    extern "Rust" {
+                        fn TRNG_IRQ();
+                    }
+                    unsafe {
+                        TRNG_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::PLL_SYS_IRQ => {
+                    extern "Rust" {
+                        fn PLL_SYS_IRQ();
+                    }
+                    unsafe {
+                        PLL_SYS_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::PLL_USB_IRQ => {
+                    extern "Rust" {
+                        fn PLL_USB_IRQ();
+                    }
+                    unsafe {
+                        PLL_USB_IRQ();
+                    }
+                }
+                rp235x_pac::Interrupt::POWMAN_IRQ_POW => {
+                    extern "Rust" {
+                        fn POWMAN_IRQ_POW();
+                    }
+                    unsafe {
+                        POWMAN_IRQ_POW();
+                    }
+                }
+                rp235x_pac::Interrupt::POWMAN_IRQ_TIMER => {
+                    extern "Rust" {
+                        fn POWMAN_IRQ_TIMER();
+                    }
+                    unsafe {
+                        POWMAN_IRQ_TIMER();
+                    }
+                }
+            }
+        }
+    }
+
+    /// Our default IRQ handler.
+    ///
+    /// Just panics.
+    ///
+    /// # Safety
+    ///
+    /// Do not call this function - it is called automatically when our
+    /// `MachineExternal` interrupt handler can't find anything better to call.
+    #[no_mangle]
+    #[allow(non_snake_case)]
+    unsafe fn DefaultIrqHandler() {
+        panic!();
```

Comment:
cortex_m does not panic but `loop {}` in the default handler.
It did panic for a short while in the past, but that was reverted here: https://github.com/rust-embedded/cortex-m-rt/pull/289/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L993

I'm not sure if the reasons for that decision also apply for RISC-V, but at least for the rp2040/rp235x case it would be nicer if both implementations did the same.

Reply:
I'll fix this

Reply:
Changed it to spin